core/src/components/tab-button/tab-button.ios.scss

```
@@ -26,6 +26,64 @@
   line-height: 16px;
 }
 
+:host ::slotted(ion-badge:empty) {
+  @include padding(0, 0);
+
+  width: $tab-ios-button-badge-size-emtpy;
+  height: $tab-ios-button-badge-size-emtpy;
+}
+
+:host ::slotted(ion-badge[vertical]) {
+  @include position(null, null, null, $tab-ios-button-badge-start-position);
+}
+
+:host ::slotted(ion-badge.badge-vertical-top) {
+  @include position($tab-ios-button-badge-top-position, null, null, null);
+}
+
+:host ::slotted(ion-badge.badge-vertical-top:empty) {
+  @include position($tab-ios-button-badge-top-position-empty, null, null, null);
+}
+
+:host ::slotted(ion-badge.badge-vertical-bottom) {
+  @include position($tab-ios-button-badge-bottom-position, null, null, null);
+}
+
+:host ::slotted(ion-badge.badge-vertical-bottom:empty) {
+  @include position($tab-ios-button-badge-bottom-position-empty, null, null, null);
+}
+
+:host ::slotted(ion-badge[vertical]:not(:empty)) {
+  @include padding(
+    $tab-ios-button-badge-padding-top,
+    $tab-ios-button-badge-padding-end,
+    $tab-ios-button-badge-padding-bottom,
+    $tab-ios-button-badge-padding-start
+  );
+
+  display: flex;
+
+  align-items: center;
+  justify-content: center;
+
+  min-width: $tab-ios-button-badge-size;
+  height: $tab-ios-button-badge-size;
+
+  font-size: $tab-ios-button-badge-font-size;
+}
+
+:host(.tab-layout-icon-bottom) ::slotted(ion-badge.badge-vertical-top) {
+  @include position($tab-ios-button-icon-bottom-badge-top-position, null, null, null);
+}
+
+:host(.tab-layout-icon-bottom) ::slotted(ion-badge.badge-vertical-bottom) {
+  @include position($tab-ios-button-icon-bottom-badge-bottom-position, null, null, null);
+}
+
+:host(.tab-layout-icon-bottom) ::slotted(ion-badge.badge-vertical-bottom:empty) {
+  @include position($tab-ios-button-icon-bottom-badge-bottom-position-empty, null, null, null);
```

Since we don't change any of the other value, we can depend on the defaults. I recommend reviewing how `position` works.

```suggestion
:host(.tab-layout-icon-bottom) ::slotted(ion-badge.badge-vertical-top) {
  @include position($tab-ios-button-icon-bottom-badge-top-position);
}

:host(.tab-layout-icon-bottom) ::slotted(ion-badge.badge-vertical-bottom) {
  @include position($tab-ios-button-icon-bottom-badge-bottom-position);
}

:host(.tab-layout-icon-bottom) ::slotted(ion-badge.badge-vertical-bottom:empty) {
  @include position($tab-ios-button-icon-bottom-badge-bottom-position-empty);
```

---

Same as before